crates/valence_anvil/src/lib.rs

```diff
@@ -86,51 +182,275 @@ impl RegionFolder {
         let region_x = pos_x.div_euclid(32);
         let region_z = pos_z.div_euclid(32);
 
-        let region = match self.regions.get_mut(&(region_x, region_z)) {
-            Some(RegionEntry::Occupied(region)) => region,
-            Some(RegionEntry::Vacant) => return Ok(None),
-            None => {
-                let path = self
-                    .region_root
-                    .join(format!("r.{region_x}.{region_z}.mca"));
-
-                let mut file = match File::options().read(true).write(true).open(path) {
-                    Ok(file) => file,
-                    Err(e) if e.kind() == ErrorKind::NotFound => {
-                        self.regions.put((region_x, region_z), RegionEntry::Vacant);
-                        return Ok(None);
+        let Some(region) =
+            Self::get_region(&mut self.regions, &self.region_root, region_x, region_z)?
+        else {
+            return Ok(None);
+        };
+
+        region.get_chunk(pos_x, pos_z, &mut self.compression_buf, &self.region_root)
+    }
+
+    /// Deletes the chunk at the given chunk position, returning whether the
+    /// chunk existed before it was deleted.
+    ///
+    /// Note that this only marks the chunk as deleted so that it cannot be
+    /// retrieved, and can be overwritten by other chunks later. It does not
+    /// decrease the size of the region file.
+    ///
+    /// This method is simple enough to not require the `write` feature.
+    pub fn delete_chunk(&mut self, pos_x: i32, pos_z: i32) -> Result<bool, RegionError> {
```

Comment:
I think it's more appropriate for an optimize method, especially since I'm not reducing the size of the file if there are any chunks left. Could leave that to another PR though.